gpm_api/tests/test_visualization/test_plot.py

```diff
@@ -0,0 +1,451 @@
+        self,
+        grid_antimeridian_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data going over the antimeridian"""
+
+        plot.plot_map(grid_antimeridian_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_grid_nan_lon(
+        self,
+        grid_nan_lon_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data with NaN longitudes"""
+
+        plot.plot_map(grid_nan_lon_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_invalid(
+        self,
+    ) -> None:
+        """Test invalid data"""
+
+        da = xr.DataArray()
+        with pytest.raises(ValueError):
+            plot.plot_map(da)
+
+
+class TestPlotImage:
+    """Test the plot_image function"""
+
+    def test_orbit(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data"""
+
+        plot.plot_image(orbit_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_grid(
+        self,
+        grid_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting grid data"""
+
+        plot.plot_image(grid_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_invalid(
+        self,
+    ) -> None:
+        """Test invalid data"""
+
+        da = xr.DataArray()
+        with pytest.raises(ValueError):
+            plot.plot_image(da)
+
+
+class TestPlotMapMesh:
+    """Test the plot_map_mesh function"""
+
+    def test_orbit(
+        self,
+        orbit_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data"""
+
+        plot.plot_map_mesh(orbit_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_antimeridian(
+        self,
+        orbit_antimeridian_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data going over the antimeridian"""
+
+        plot.plot_map_mesh(orbit_antimeridian_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_antimeridian_projection(
+        self,
+        orbit_antimeridian_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data going over the antimeridian on orthographic projection"""
+
+        crs_proj = ccrs.Orthographic(180, 0)
+        plot.plot_map_mesh(orbit_antimeridian_dataarray, subplot_kwargs={"projection": crs_proj})
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_pole(
+        self,
+        orbit_pole_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data going over the south pole"""
+
+        plot.plot_map_mesh(orbit_pole_dataarray)
+        save_and_check_figure(get_test_name(self))
+
+    def test_orbit_pole_projection(
+        self,
+        orbit_pole_dataarray: xr.DataArray,
+    ) -> None:
+        """Test plotting orbit data going over the south pole on orthographic projection"""
+
+        crs_proj = ccrs.Orthographic(0, -30)
```

Comment:
Why centered on -30? If SouthPole should be `ccrs.Orthographic(180, -90)` no? 
Or alternatively `ccrs.SouthPolarStereo()`